.pre-commit-config.yaml

```diff
@@ -22,6 +22,11 @@ repos:
       - id: flake8
         additional_dependencies: [flake8-2020]
 
+  - repo: https://github.com/astral-sh/uv-pre-commit
+    rev: 0.5.20
+    hooks:
+      - id: uv-lock
+
```

Comment:
Does that mean pre-commit will fail every time there's a dependency update? If yes, I'd prefer to stick with periodic manual updates as we do now.

Reply:
This will fail every time `uv.lock` is out of sync with packages defined in `pyproject.toml`.

Dependabot [doesn't support](https://docs.astral.sh/uv/guides/integration/dependency-bots/#dependabot) uv yet. I dunno if it will continue opening PRs for security issues after requirements are listed in `pyproject.toml`.

Reply:
This looks good if running pre-commit won't update dependencies unless we modify pyproject.toml. :+1:

(Regarding Dependabot, it's OK; we don't really need Dependabot for our Python development dependencies.)